### PR TITLE
fix DeleteChats and self destruction messages

### LINKED ISSUE #360: [BUG] Regression - Delete a Conversation is Broken
### Description

Prior to version 0.9.7, it was possible to delete a conversation. Now nothing happens when you select a conversation and hit Trash icon. 

### Steps to Reproduce

1. Go to the list of conversations
2. Long press on the conversation to make sure it is selected
3. Tap Trash icon on the top

**Expected behavior:** [What you expect to happen]

The conversation should disappear from the list

**Actual behavior:** [What actually happens]

The conversation gets deselected and remains in the list.

#### Versions

Please provide the Version as written in `Settings->About Axolotl`
0.9.7

#### Device
What device or OS are you using?
Volla Phone + UT

#### Link to Debug Log
Please provide a link to debug from ~/.cache/upstart/application-click-textsecure.nanuc_textsecure*.log
Be careful it contains sensible data

### PR DESCRIPTION
**Pull request by nanu-c**
This closes #360, #334 and #167. Maybe also #279.

Attachments are now also deleted, when a conversation or a selfdestructiong message is removed.

### CODE REVIEW

File: app/store/message.go [hunk 1]
```diff
@@ -118,11 +118,16 @@ func LoadMessagesFromDB() error {
...
 }
+
 func DeleteMessage(id int64) error {
-	// TODO: delete Attachments
-	_, err := DS.Dbx.Exec("DELETE FROM messages WHERE id = ?", id)
+	err := deleteAttachmentForMessage(id)
+	if err!=nil{
+		log.Errorln("[axolotl] could not delete attachment", err)
+	}
+	_, err = DS.Dbx.Exec("DELETE FROM messages WHERE id = ?", id)
```
Comment (Flaburgan) [inline app/store/message.go:127]: Do we want to continue to delete the message if the deletion of the attachment failed?

Comment (nanu-c) [reply app/store/message.go:127]: When we delete the message, we loose all reference to the attachment, so maybe not

Comment (Flaburgan) [reply app/store/message.go:127]: Yeah so let's completely fail here please.

**Comment by Flaburgan** [comment]
Thank you! :cookie:

**Comment by Flaburgan** [comment]
Thank you! :cookie:

**Pull request merged**